recipe/bld.bat

```diff
@@ -166,30 +207,39 @@ if errorlevel 1 exit 1
 rd /s /q %PREFIX%\Lib\lib2to3\tests\
 if errorlevel 1 exit 1
 
-:: We need our Python to be found!
+:: We need our Python to be found
 if "%_D%" neq "" copy %PREFIX%\python%_D%.exe %PREFIX%\python.exe
+if "%_D%" neq "" copy %PREFIX%\python%_D%.pdb %PREFIX%\python.pdb
 
 %PREFIX%\python.exe -Wi %PREFIX%\Lib\compileall.py -f -q -x "bad_coding|badsyntax|py2_" %PREFIX%\Lib
 if errorlevel 1 exit 1
 
 :: Pickle lib2to3 Grammar
 %PREFIX%\python.exe -m lib2to3 --help
 
-:: Some quick tests for common failures
-echo "Testing print() does not print: Hello"
-%CONDA_EXE% run -p %PREFIX% cd %PREFIX% & %PREFIX%\python.exe -c "print()" 2>&1 | findstr /r /c:"Hello"
+echo CONDA_EXE is %CONDA_EXE%
+echo where conda is
+where conda
+echo where python is
+where python
+
+echo "Testing print() does not print Hello"
+conda run -p %PREFIX% python -c "print()" 2>&1 | findstr /r /c:"Hello"
 if %errorlevel% neq 1 exit /b 1
 
-echo "Testing print('Hello') prints: Hello"
-%CONDA_EXE% run -p %PREFIX% cd %PREFIX% & %PREFIX%\python.exe "print('Hello')" 2>&1 | findstr /r /c:"Hello"
+echo "Testing print('Hello') prints Hello"
+conda run -p %PREFIX% python -c "print('Hello')" 2>&1 | findstr /r /c:"Hello"
 if %errorlevel% neq 0 exit /b 1
 
-echo "Testing import of os (no DLL needed) does not print: The specified module could not be found"
-%CONDA_EXE% run -p %PREFIX% cd %PREFIX% & %PREFIX%\python.exe -v -c "import os" 2>&1
-%CONDA_EXE% run -p %PREFIX% cd %PREFIX% & %PREFIX%\python.exe -v -c "import os" 2>&1 | findstr /r /c:"The specified module could not be found"
+echo "Testing import of os does not print The specified module could not be found"
+conda run -p %PREFIX% python -v -c "import os" 2>&1
+conda run -p %PREFIX% python -v -c "import os" 2>&1 | findstr /r /c:"The specified module could not be found"
```

Comment:
I am not sure I understand why the variable names are replaced

Reply:
Because %CONDA_EXE% is not what should be run by anyone, ever, including scripts.

conda is, on Windows, only ever allowed to be a shell function or script (a batch file or a powershell functions). It is critial for its correct behaviour that this is the case. I could go into the details but it's down to the fact that the scripts/batch files correctly prefix PATH with the 6 entries needed for the correct execution of the real conda.exe before calling it.